src/pages/LeaderboardPage/components/LeaderboardContainer.tsx

```diff
@@ -1,94 +1,121 @@
   const [activeCompetitionIndex, setActiveCompetitionIndex] = useState(0);
-  const accounts = useLeaderboardRankedAccounts();
+
   const leaderboardPageKey = useLeaderboardPageKey();
-  const isLoading = !accounts;
-  const accountsStruct = useMemo(
-    () => ({
-      isLoading,
-      data: accounts ? accounts : [],
-      error: null,
-      updatedAt: 0,
-    }),
-    [accounts, isLoading]
-  );
+
   const { chainId } = useChainId();
   const { active } = useWallet();
 
   const page = LEADERBOARD_PAGES[leaderboardPageKey];
 
-  const [, setLeaderboardType] = useLeaderboardTypeState();
+  const [, setLeaderboardTimeframeType] = useLeaderboardTimeframeTypeState();
+  const [leaderboardDataType, setLeaderboardDataType] = useLeaderboardDataTypeState();
+
+  const competitionLabels = useMemo(() => [t`Top PnL ($)`, t`Top PnL (%)`], []);
+  const leaderboardTimeframeLabels = useMemo(() => [t`Total`, t`Last 30 days`, t`Last 7 days`], []);
+  const leaderboardDataTypeLabels = useMemo(() => [t`Top Addresses`, t`Top Positions`], []);
 
   const activeCompetition: CompetitionType | undefined = isCompetition
     ? activeCompetitionIndex === 0
       ? "notionalPnl"
       : "pnlPercentage"
     : undefined;
 
-  const handleLeaderboardTabChange = useCallback(
-    (index: number) => setActiveLeaderboardIndex(index),
-    [setActiveLeaderboardIndex]
+  const handleLeaderboardTimeframeTabChange = useCallback(
+    (index: number) => setActiveLeaderboardTimeframeIndex(index),
+    [setActiveLeaderboardTimeframeIndex]
   );
   const handleCompetitionTabChange = useCallback(
     (index: number) => setActiveCompetitionIndex(index),
     [setActiveCompetitionIndex]
   );
 
+  const handleLeaderboardDataTypeTabChange = useCallback(
+    (index: number) => setActiveLeaderboardDataTypeIndex(index),
+    []
+  );
+
   const pageKey = useLeaderboardPageKey();
   const leaderboardChainId = useLeaderboardChainId();
 
   useEffect(() => {
-    setActiveLeaderboardIndex(0);
+    setActiveLeaderboardTimeframeIndex(0);
     setActiveCompetitionIndex(0);
   }, [pageKey]);
 
   useEffect(() => {
-    if (activeLeaderboardIndex === 0) {
-      setLeaderboardType("all");
-    } else if (activeLeaderboardIndex === 1) {
-      setLeaderboardType("30days");
+    if (activeLeaderboardTimeframeIndex === 0) {
+      setLeaderboardTimeframeType("all");
+    } else if (activeLeaderboardTimeframeIndex === 1) {
+      setLeaderboardTimeframeType("30days");
     } else {
-      setLeaderboardType("7days");
+      setLeaderboardTimeframeType("7days");
     }
-  }, [activeLeaderboardIndex, setLeaderboardType]);
+  }, [activeLeaderboardTimeframeIndex, setLeaderboardTimeframeType]);
+
+  useEffect(() => {
+    if (activeLeaderboardDataTypeIndex === 0) {
+      setLeaderboardDataType("accounts");
+    } else {
+      setLeaderboardDataType("positions");
+    }
+  }, [activeLeaderboardDataTypeIndex, setLeaderboardDataType]);
+
+  const title = useMemo(() => {
```

Comment:
this looks very similar to getLabel function, defined later, are they different? or the  getLabel could be used here?